src/Ethereum/EIP4337.cpp

```diff
@@ -60,4 +59,31 @@ std::string getEIP4337DeploymentAddress(const std::string& factoryAddress, const
     return Ethereum::checksumed(Address(hexEncoded(create2Address(factoryAddress, salt, initCodeHash))));
 }
 
+std::string getEIP4337AddressFromOwnerBytes(const std::string& factoryAddress, const std::string& bytecode, const std::string& diamondCutFacetAddress, const std::string& accountFacetAddress, const std::string& verificationFacetAddress, const std::string& entryPointAddress, const std::string& securityManagerAddress, const std::string& facetRegistryAddress, const std::string& ownerPublicKey) {
```

Comment:
This also could be refactored. At least, by adding `struct EIP4337AddressFromOwnerParams { ... };` locally. Or by using Protobuf if this way has been chosen above.

Reply:
Can `struct EIP4337AddressFromOwnerParams { ... };` be accessed and built from Swift/Kotlin clients?

Reply:
If it's declared in any file within`wallet-core/src` directory, it won't be accessed from other langs.
So there should be `wallet-core/include/TWEIP4337AddressFromOwnerParams.h`.
Probably, it could follow the same way as `TWPrivateKey`:
1. Forward declaration:
https://github.com/trustwallet/wallet-core/blob/d67078daa580b37063c97be66a625aaee9664882/include/TrustWalletCore/TWPrivateKey.h#L17-L19
2. Method declaration:
https://github.com/trustwallet/wallet-core/blob/d67078daa580b37063c97be66a625aaee9664882/include/TrustWalletCore/TWPrivateKey.h#L23-L28
3. Actual declaration:
https://github.com/trustwallet/wallet-core/blob/d67078daa580b37063c97be66a625aaee9664882/src/PrivateKey.h#L92C1-L95

@hewigovens what do you think?

Reply:
The most recent exemple we have following the protobuf approach is liquid staking:

```cpp
// Copyright © 2017-2023 Trust Wallet.
//
// This file is part of Trust. The full Trust copyright notice, including
// terms governing use, modification, and redistribution, is contained in the
// file LICENSE at the root of the source code distribution tree.
#pragma once

#include "TWBase.h"
#include "TWData.h"
#include "TWString.h"

TW_EXTERN_C_BEGIN

/// THORChain swap functions
TW_EXPORT_STRUCT
struct TWLiquidStaking;

/// Builds a LiquidStaking transaction input.
///
/// \param input The serialized data of LiquidStakingInput.
/// \return The serialized data of LiquidStakingOutput.
TW_EXPORT_STATIC_METHOD
TWData *_Nonnull TWLiquidStakingBuildRequest(TWData *_Nonnull input);

TW_EXTERN_C_END
```

cc @rsrbk